internal/runtime/client/client.go

```diff
@@ -331,15 +377,31 @@ func (c *client) CallExtension(ctx context.Context, hook runtimecatalog.Hook, fo
 	// Prepare the request by merging the settings in the registration with the settings in the request.
 	request = cloneAndAddSettings(request, registration.Settings)
 
-	opts := &httpCallOptions{
+	var cacheKey string
+	if options.WithCaching {
+		// Return a cached response if response is cached.
+		cacheKey = options.CacheKeyFunc(registration, request)
+		if cacheEntry, ok := options.Cache.Has(cacheKey); ok {
+			// Set response to cacheEntry.Response.
+			outVal := reflect.ValueOf(response)
+			cacheVal := reflect.ValueOf(cacheEntry.Response)
+			if !cacheVal.Type().AssignableTo(outVal.Type()) {
+				return fmt.Errorf("failed to call extension handler %q: cached response of type %s instead of type %s", name, cacheVal.Type(), outVal.Type())
+			}
+			reflect.Indirect(outVal).Set(reflect.Indirect(cacheVal))
```

Comment:
```suggestion
			response = cacheEntry.Response
```

Why can't it be simply this?

Reply:
I mean, don't we use generics just to not need that reflect stuff?

Reply:
Setting the response like this doesn't propagate back up to the call site

Reply:
I think generic have nothing to do with this code here :)

Reply:
Usually in all the places (CAPI & CR) where we pass "output" variables via input variables we do one of:
* only modify sub-fields of the passed in variable
* use some sort of json.Unmarshal or Decode to write the values of the field

The alternative to reflection here is probably taking the cacheEntry.Reponse and using Marhsal | Unmarshal to set the fields in response

Reply:
Interesting. Thanks for explaining

Reply:
That being said. Open to suggestions if you know/find an alternative that works :)

Reply:
Inside httpCall we use `opts.catalog.Convert`

But I think yours is better for the caching use-case.